x-pack/apm-server/sampling/eventstorage/storage.go

```diff
@@ -185,30 +172,70 @@ func (rw *ReadWriter) WriteTraceEvent(traceID string, id string, event *modelpb.
 
 func (rw *ReadWriter) writeEntry(e *badger.Entry, opts WriterOpts) error {
 	rw.pendingWrites++
-	err := rw.txn.SetEntry(e.WithTTL(opts.TTL))
-	// Attempt to flush if there are 200 or more uncommitted writes.
-	// This ensures calls to ReadTraceEvents are not slowed down;
-	// ReadTraceEvents uses an iterator, which must sort all keys
-	// of uncommitted writes.
-	// The 200 value yielded a good balance between read and write speed:
-	// https://github.com/elastic/apm-server/pull/8407#issuecomment-1162994643
+	entrySize := estimateSize(e)
+	lsm, vlog := rw.s.db.Size()
```

Comment:
Add a comment here to mention Size() is updated in the background every minute

Reply:
we had a comment like that in https://github.com/elastic/apm-server/blob/v8.11.4/x-pack/apm-server/sampling/eventstorage/storage.go#L77 and it helps readers understand why we don't really care about races in storage limit checking in this file.